### PR TITLE
Fix up

### PR DESCRIPTION
**Pull request by JingLin0**
# Description of change
(write a short description here or paste a link to JIRA)

# QA steps
 - [ ] automated tests passing
 - [ ] manual qa steps passing (list below)
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,100 +1,5 @@
...
-config.json
-.autoenv.zsh
-
-*~
\ No newline at end of file
+Pipfile*
+catalog.json
+.vscode/
+*.ndjson
+config.json
```
Comment (paddie) [inline .gitignore:105]: actually, keep most of the default .gitignore file here, it handles a lot of default things, like ignored `.DS_Store` files for Mac users etc. along with ay locally installed pip packages that can be passed in like `.eggs` files etc. Feel free to add the stuff you need to it, and delete the stuff you don't need like anything after the `# Custom stuff`

File: tap_salesforce/__init__.py [hunk 2]
```diff
@@ -125,7 +125,17 @@ def do_discover(sf):
...
+    objects_to_discover = [
+        "Account",
+        "Contact",
+        "Lead",
+        "Opportunity",
+        "Campaign",
+        "AccountContactRelation",
+        "AccountContactRole",
+        "OpportunityContactRole",
+    ]
```
Comment (paddie) [inline tap_salesforce/__init__.py:141]: So, I'm assuming these are the only ones we really care about then?

File: tap_salesforce/salesforce/__init__.py [hunk 3]
```diff
@@ -248,7 +248,7 @@ def __init__(
         self.rest_requests_attempted = 0
         self.jobs_completed = 0
         self.login_timer = None
-        self.data_url = "{}/services/data/v41.0/{}"
+        self.data_url = "{}/services/data/v48.0/{}"
```
Comment (paddie) [inline tap_salesforce/salesforce/__init__.py:251]: maybe add a global `API_VERSION` variable to make it clear that they are all using the same version.. Except if they are not of course?

File: tap_salesforce/sync.py [hunk 4]
```diff
@@ -152,6 +152,11 @@ def sync_records(sf, catalog_entry, state, counter):
     LOGGER.info("Syncing Salesforce data for stream %s", stream)
 
     for rec in sf.query(catalog_entry, state):
+        replication_key_value = replication_key and singer_utils.strptime_with_tz(
+            rec[replication_key]
+        )
+        if replication_key_value < chunked_bookmark:
```
Comment (paddie) [inline tap_salesforce/sync.py]: should it be `<=` to not perform the checkpoint-day twice?

File: tap_salesforce/salesforce/__init__.py [hunk 5]
```diff
@@ -305,12 +305,12 @@ def _make_request(
...
-            LOGGER.info(
-                "Making %s request to %s with params: %s", http_method, url, params
-            )
+            # LOGGER.info(
+            #     "Making %s request to %s with params: %s", http_method, url, params
+            # )
             resp = self.session.get(url, headers=headers, stream=stream, params=params)
         elif http_method == "POST":
-            LOGGER.info("Making %s request to %s with body %s", http_method, url, body)
+            # LOGGER.info("Making %s request to %s with body %s", http_method, url, body)
```
Comment (paddie) [inline tap_salesforce/salesforce/__init__.py]: either remove or set log-level to ignore info messages

File: tap_salesforce/__init__.py [hunk 6]
```diff
@@ -353,7 +356,7 @@ def do_sync(sf, catalog, state):
         )
 
         catalog_metadata = metadata.to_map(catalog_entry["metadata"])
-        replication_key = catalog_metadata.get((), {}).get("replication-key")
+        replication_key = catalog_metadata.get((), {}).get("valid-replication-keys")[0]
```
Comment (paddie) [inline tap_salesforce/__init__.py:359]: I love the commit message :)

**Review by paddie — COMMENTED** [review]
looks good, excellent commit structure! I'm only worried about the one `<` check for the checkpoint, but I might be wrong.

File: tap_salesforce/__init__.py (hunk 2 above)
Comment (JingLin0) [reply tap_salesforce/__init__.py:141]: yes

File: tap_salesforce/sync.py (hunk 4 above)
Comment (JingLin0) [reply tap_salesforce/sync.py]: you are right

File: tap_salesforce/__init__.py (hunk 2 above)
Comment (przywartya) [reply tap_salesforce/__init__.py:141]: How can I know that I care about an object? Who knows that?

File: tap_salesforce/__init__.py (hunk 6 above)
Comment (przywartya) [reply tap_salesforce/__init__.py:359]: But I don't understand, are we making it DRY or is it just an irony? :D

Comment (przywartya) [reply tap_salesforce/__init__.py:359]: If think it wouldn't hurt us to create a function doing this `catalog_metadata.get((), {}).get("valid-replication-keys")[0]`

Comment (paddie) [reply tap_salesforce/__init__.py:359]: she's making it DRY later (not in this commit) - right now she is just fixing it

File: .gitignore [hunk 7]
```diff
@@ -97,4 +97,9 @@ env.sh
 config.json
 .autoenv.zsh
 
-*~
\ No newline at end of file
+*~
+Pipfile*
```
Comment (przywartya) [inline .gitignore:101]: @paddie 
Why do we keep Pipfiles out of our repositories?

File: tap_salesforce/__init__.py (hunk 2 above)
Comment (JingLin0) [reply tap_salesforce/__init__.py:141]: check out the salesforce story card on trello

File: tap_salesforce/salesforce/__init__.py (hunk 3 above)
Comment (JingLin0) [reply tap_salesforce/salesforce/__init__.py:251]: but these three api-version changes are in three different files, they can't use one global variables

File: .gitignore (hunk 7 above)
Comment (paddie) [reply .gitignore:101]: Ha, good question. Right now, we do it such that every time we execute it, it gets the latest version of the dependency. I guess we could actually include them and just not lock the version at all, but we can talk about it.

File: tap_salesforce/salesforce/__init__.py (hunk 3 above)
Comment (przywartya) [reply tap_salesforce/salesforce/__init__.py:251]: they can, you can import variable form some file in each 👍

**Review by przywartya — APPROVED** [review]

File: tap_salesforce/salesforce/__init__.py (hunk 3 above)
Comment (paddie) [reply tap_salesforce/salesforce/__init__.py:251]: yeah, or be an argument to some overlying function thing that we define in the entry-function. But the pain is not great for just 3 files.

**Review by paddie — APPROVED** [review]

**Pull request merged**